backend/code_review_backend/issues/tests/test_stats.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import hashlib
+import random
+import uuid
+
+from django.contrib.auth.models import User
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from code_review_backend.issues.models import Issue
+from code_review_backend.issues.models import Repository
+
+
+class StatsAPITestCase(APITestCase):
+    def setUp(self):
+        # Create a user
+        self.user = User.objects.create(username="crash_user")
+
+        # Create a repo
+        self.repo = Repository.objects.create(
+            id=1, phid="PHID-REPO-xxx", slug="myrepo", url="http://repo.test/myrepo"
+        )
+
+        # Create a revision
+        revision = self.repo.revisions.create(
+            id=10, phid=f"PHID-DREV-arev", title=f"Revision A", bugzilla_id=None
+        )
+
+        # Create some diffs
+        for i in range(10):
+            revision.diffs.create(
+                id=i + 1,
+                phid=f"PHID-DIFF-{i+1}",
+                review_task_id=f"task-{i}",
+                mercurial_hash=hashlib.sha1(f"hg {i}".encode("utf-8")).hexdigest(),
+            )
+
+        # Create lots of issues randomly affected on diffs
+        analyzers = ["analyzer-X", "analyzer-Y", "analyzer-Z"]
+        checks = ["check-1", "check-10", "check-42", "check-1000", None]
+
+        for i in range(500):
+            Issue.objects.create(
+                diff_id=random.randint(1, 10),
```

Comment:
Can you add a few Issues with no linked diff too, just to make sure that works too?

Reply:
Done